### PR TITLE
add linters and fix findings

### LINKED ISSUE #368: Enhance the Golang linting
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
Recently `.golangci.yml` got added via https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/367 with timeout added, need more checks added into the linting, refer https://github.com/kubernetes-sigs/cluster-api/blob/main/.golangci.yml file.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Add linters and fix findings

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #368 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
add linters and fix findings
```

### CODE REVIEW

**Review by Amulyam24 — APPROVED** [review]
/lgtm Thanks @Prajyot-Parab!

File: controllers/suite_test.go [hunk 1]
```diff
@@ -20,18 +20,17 @@ import (
...
 	"k8s.io/client-go/rest"
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 
-	infrastructurev1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
-
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
```
Comment (mkumatag) [inline controllers/suite_test.go:33]: wondering if there a way to allow dot(.) imports in the end, if not - wondering if we can ignore warning/error message for the dot import(prefer having them in the end) easy to see what imports are dot in different section.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: .golangci.yml [hunk 2]
```diff
@@ -1,16 +1,208 @@
...
+        alias: capiv1alpha4
+      - pkg: sigs.k8s.io/cluster-api/api/v1beta1
+        alias: capiv1beta1
+      # CAPI-IBMCLOUD
+      - pkg: sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1alpha3
+        alias: clusterv1alpha3
+      - pkg: sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1alpha4
+        alias: clusterv1alpha4
+      - pkg: sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1
+        alias: clusterv1beta1
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply .golangci.yml]: renamed to `infra*`

File: cloud/scope/powervs_image.go [hunk 3]
```diff
@@ -137,6 +136,7 @@ func NewPowerVSImageScope(params PowerVSImageScopeParams) (scope *PowerVSImageSc
 	}, nil
 }
 
+// SetInstanceState will set the state for the machine.
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/powervs_image.go]: Fixed.

File: .golangci.yml [hunk 4]
```diff
@@ -1,16 +1,208 @@
...
 
 run:
   timeout: 10m
+  build-tags:
+  - tools
+  - e2e
+  skip-files:
+  - "zz_generated.*\\.go$"
+  skip-dirs:
+  - third_party
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply .golangci.yml]: Removed.

File: controllers/ibmpowervscluster_controller.go [hunk 5]
```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Package controllers implements controllers code.
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply controllers/ibmpowervscluster_controller.go]: Added `doc.go` to packages and fixed.

File: controllers/suite_test.go (hunk 1 above)
Comment (Prajyot-Parab) [reply controllers/suite_test.go]: Added exception and fixed.

File: pkg/cloud/services/authenticator/authenticator.go [hunk 6]
```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Package authenticator implements authenticator code.
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply pkg/cloud/services/authenticator/authenticator.go]: Added `doc.go` to packages and fixed.

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**